packages/sanity/src/core/form/inputs/files/ImageInput/usePreviewImageSource.ts

```diff
@@ -0,0 +1,62 @@
+import {isImageSource} from '@sanity/asset-utils'
+import {type CSSProperties, useMemo} from 'react'
+import {useDevicePixelRatio} from 'use-device-pixel-ratio'
+
+import {type ImageUrlBuilder} from '../types'
+import {type BaseImageInputValue} from './types'
+
+type Dimensions = [width: number, height: number]
+
+/*
+  Used for setting the initial image height - specifically for images
+  that are small and so can take less space in the document
+*/
+const getImageSize = (src: string): Dimensions => {
+  const imageUrlParams = new URLSearchParams(src.split('?')[1])
+  const rect = imageUrlParams.get('rect')
+
+  if (rect) {
+    return [rect.split(',')[2], rect.split(',')[3]].map(Number) as Dimensions
+  }
+
+  return src.split('-')[1].split('.')[0].split('x').map(Number) as Dimensions
+}
```

Comment:
I realize this is old code from `ImagePreview.tsx`, but this will break when using a dataset with a dash in the name, as it splits on `-`. The `rect` handling still may need some manual extraction such as `new URL(src).searchParams.get('rect')`, but the dimensions of the image should preferably use something like [`@sanity/asset-utils#getImageDimensions`](https://sanity-io.github.io/asset-utils/index.html#getimagedimensions).

Reply:
Ah, good catch! Should have scrutinised this code a bit more before copy/pasting it. Now using `getImageDimensions` instead. Thanks!